src/worker.ts

```diff
@@ -0,0 +1,75 @@
+import { Value } from "@sinclair/typebox/value";
+import { run } from "./run";
+import { Env, envConfigValidator } from "./types/env";
+import { assistivePricingSettingsSchema } from "./types/plugin-input";
+
+export default {
+  async fetch(request: Request, env: Env): Promise<Response> {
+    try {
+      if (request.method !== "POST") {
+        return new Response(JSON.stringify({ error: `Only POST requests are supported.` }), {
+          status: 405,
+          headers: { "content-type": "application/json", Allow: "POST" },
+        });
+      }
+      const contentType = request.headers.get("content-type");
+      if (contentType !== "application/json") {
+        return new Response(JSON.stringify({ error: `Error: ${contentType} is not a valid content type` }), {
+          status: 400,
+          headers: { "content-type": "application/json" },
+        });
+      }
+      if (!envConfigValidator.test(env)) {
+        const errorDetails: string[] = [];
+        for (const error of envConfigValidator.errors(env)) {
+          errorDetails.push(`${error.path}: ${error.message}`);
+        }
+        return new Response(JSON.stringify({ error: `The environment is invalid: ${errorDetails.join("; ")}` }), {
+          status: 400,
+          headers: { "content-type": "application/json" },
+        });
+      }
+      const webhookPayload = await request.json();
+      const signature = webhookPayload.signature;
+      delete webhookPayload.signature;
+      if (!(await verifySignature(env.UBIQUIBOT_PUBLIC_KEY, webhookPayload, signature))) {
+        return new Response(JSON.stringify({ error: `Error: Signature verification failed` }), {
```

Comment:
Errors should be 5xx class 

This is not an error. It is forbidden. We should change the language here.

Reply:
Let's replace `Error` with `Forbidden`, with `403` status.